### ISSUE TITLE
timestamp in log only

### ISSUE THREAD

**Issue by williamblair333** [WoodenNautilus/tabby-serial-timestamp#1]
Is it possible to configure the timestamp to only show up in the log and not live output?

**Comment by WoodenNautilus** [comment]
That's a good question. I will have to dig further into tabby-serial to do that. I will let you know if I find a solution.
Also, if you have any knowledge in computer science, don't bother looking into the code !

What do you use to save the serial session to a file ? The plugin "save-output" ?

**Comment by williamblair333** [comment]
Apologies for long delay.  If you have a dummies / hand holding / getting started guide to tabby plugins, I'll definitely take a look and see if I can be useful.  Reply here or send an email to my username at gmail

I use the "save output to a file" option when I right-click the desired tab.  I assume "save-output" is that plugin as none of the other plugin names seem relevant. 

Thanks for all your hard work and hopefully I can contribute something.

**Comment by WoodenNautilus** [comment]
Hello,

If you want to understand how the plugins are created you can refer to [this file](https://github.com/Eugeny/tabby/blob/master/HACKING.md). The creators of Tabby did a great job at getting you started.

For the issue reported, I do not know if there is a solution. Serial-Timestamp is a SessionMiddleware which means it takes data from the Session and then send it to the Terminal. There is no communication between Serial-Timestamp and Save-Output, other than the data processed. Thus, no way of saving the data with the date but displaying it without.

Unless @Eugeny has a better idea, having the date to only show-up in the logs would probably means porting the save-output code in the serial-timestamp plugin.

If this is something which fair amount of people are interested in, I could consider developing this.

**Comment by Eugeny** [comment]
Agreed. Feeding different output to save-output and the terminal would require support from the save-output plugin itself, there's nothing this plugin can do to make it work

**Comment by WoodenNautilus** [comment]
Let's say I would be ready to undergo these developments. What do you reckon is the best solution ? Adding an extra output to "save-output" ? Or porting what save-output does into serial-timestamp ?

**Comment by Eugeny** [comment]
I've been thinking about this and I can't figure out a good way to do this, the only idea that comes to mind is to output the timestamp wrapped in a custom escape sequence, e.g. `"\x1b]1337;Timestamp=something\x07"`, which will not show up on screen, but will still be seen and can be processed by `save-output`
